### PR TITLE
Botany composting & shovel/spade (or, make botany sustainable)

### PR DESCRIPTION
**Pull request by 20kdc**
+ Composting
+ Shovel/Spade
+ Add a bucket & spade to Saltern's Botany
(the following is a joke: TODO: Sandcastles)

### CODE REVIEW

File: Content.Server/GameObjects/Components/Botany/PlantHolderComponent.cs [hunk 1]
```diff
@@ -764,6 +780,26 @@ public async Task<bool> InteractUsing(InteractUsingEventArgs eventArgs)
...
+                user.PopupMessageOtherClients(Loc.GetString("{0:TheName} composts {1:theName} into {2:theName}.", user, usingItem, Owner));
+
+                if (usingItem.TryGetComponent(out SolutionContainerComponent? solution2))
+                {
+                    // This deliberately discards overfill.
+                    _solutionContainer?.TryAddSolution(solution2.SplitSolution(solution2.Solution.TotalVolume));
+
+                    SkipAging++; // We're forcing an update cycle, so one age hasn't passed.
+                    ForceUpdate = true;
+                    Update();
```
Comment (PaulRitter) [inline Content.Server/GameObjects/Components/Botany/PlantHolderComponent.cs:795]: so if i understand correctly, composting just forces an additional updateloop?

File: Resources/Prototypes/Entities/Objects/Tools/botany_tools.yml [hunk 2]
```diff
@@ -98,3 +98,32 @@
...
+
+- type: entity
+  name: shovel
+  parent: BaseItem
+  id: Shovel
+  description: A large tool for digging and moving dirt.
+  components:
+    - type: Sprite
+      sprite: Constructible/Hydroponics/hydro_tools.rsi
+      state: spade
```
Comment (PaulRitter) [inline Resources/Prototypes/Entities/Objects/Tools/botany_tools.yml:124]: shovel having an identitycrisis?

**Review by PaulRitter — CHANGES_REQUESTED** [review]

File: Resources/Prototypes/Entities/Objects/Tools/botany_tools.yml (hunk 2 above)
Comment (20kdc) [reply Resources/Prototypes/Entities/Objects/Tools/botany_tools.yml:124]: 20kdc forgetting to change something during a copy/paste, more like

File: Content.Server/GameObjects/Components/Botany/PlantHolderComponent.cs (hunk 1 above)
Comment (20kdc) [reply Content.Server/GameObjects/Components/Botany/PlantHolderComponent.cs:795]: see TryAddSolution. Note that this is copy/pasted (I know, I know ;.;) from the other code which does the same thing.

**Review by PaulRitter — APPROVED** [review]

**Pull request merged**